### PR TITLE
Separate search results page

### LINKED ISSUE #73: Have search results on separate page
We only get a drop-down with results. It would be nice to also have a results page if a user types in a search and hits return and a new page will display the results in a little better fashion than the drop-down.

### PR DESCRIPTION
**Pull request by Alqanar**
Closes #73

A separate page has been added to display search results. It also allows you to search the documentation, but only within the version that was selected before going to this page.

https://user-images.githubusercontent.com/41822696/181783382-3d056326-0c1c-4b15-b310-c82773a6286b.mov

### CODE REVIEW

File: components/Search/utils.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    storeSearchResults({ query, results, version });
+
+    return results;
+  } catch (e) {
+    console.error(e);
+  }
+};
+
+export const getEmptyNotice = (query: string): string => {
+  return `No results were found for the search '${query}'`;
```
Comment (hatched) [inline components/Search/utils.ts:44]:
```suggestion
  return `No results were found for the search: '${query}'`;
```

File: components/Search/Search.tsx [hunk 2]
```diff
@@ -1,27 +1,30 @@
...
-    const rawResponse = await fetch(`/docs/api/search/?query=${query}`, {
-      method: "GET",
-    });
-
-    const response = await rawResponse.json();
-    return response.map((res) => ({ ...res, label: res.title }));
-  } catch (e) {
-    console.error(e);
-  }
+const ITEM_LINK = {
```
Comment (hatched) [inline components/Search/Search.tsx:19]: I think this should be of type `SearchResultRecord`?

File: components/Search/SeparateSearch.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
+import { useCallback, useEffect, useState } from "react";
+import { unpackSearchResults } from "utils/general";
+import { getEmptyNotice, getSearchResults } from "./utils";
+import { ProductItem } from "./ProductItem";
+import type { SearchResultRecord } from "./types";
+import styles from "./SeparateSearch.module.css";
+
+const SEARCH_THRESHOLD = 1000;
```
Comment (hatched) [inline components/Search/SeparateSearch.tsx:8]:
```suggestion
const SEARCH_TIMEOUT = 1000;
```

File: components/Search/SeparateSearch.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+export default function SeparateSearch() {
+  const [savedSearchResults] = useState(() => unpackSearchResults());
```
Comment (hatched) [inline components/Search/SeparateSearch.tsx:11]: This wasn't picking up the type from the `unpackSearchResults` for me so maybe just adding the type here...
```suggestion
  const [savedSearchResults] = useState<undefined | SearchResultMeta>(() => unpackSearchResults());
```

**Review by hatched — APPROVED** [review]

**Review by avatus — DISMISSED** [review]
Not a blocker, but can we have the `return` key send us to the search results page as well? It currently doesn't do anything (that I've noticed). Otherwise, LGTM! 👽

**Review by avatus — APPROVED** [review]

**Pull request merged**